examples/example-xds/src/main/java/io/grpc/examples/helloworldxds/HelloWorldClientXds.java

```diff
@@ -0,0 +1,110 @@
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.StatusRuntimeException;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.netty.GrpcSslContexts;
+import io.grpc.netty.NettyChannelBuilder;
+import io.grpc.xds.XdsChannelCredentials;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.net.ssl.SSLException;
+
+/**
+ * A simple xDS client that requests a greeting from the {@link HelloWorldServerXds}.
+ */
+public class HelloWorldClientXds {
+  private static final Logger logger = Logger.getLogger(HelloWorldClientXds.class.getName());
+  private final ManagedChannel channel;
+  private final GreeterGrpc.GreeterBlockingStub blockingStub;
+
+  /** Construct client connecting to HelloWorld server at {@code host:port}. */
+  public HelloWorldClientXds(String target, boolean useXdsCreds) throws SSLException {
+    this(
+        Grpc.newChannelBuilder(
+                target,
+                useXdsCreds
+                    ? XdsChannelCredentials.create(InsecureChannelCredentials.create())
+                    : InsecureChannelCredentials.create())
+            .build());
+  }
+
+  /** Construct client for accessing HelloWorld server using the channel passed. */
+  HelloWorldClientXds(ManagedChannel channel) {
+    this.channel = channel;
+    blockingStub = GreeterGrpc.newBlockingStub(channel);
+  }
+
+  public void shutdown() throws InterruptedException {
+    channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+  }
+
+  /** Say hello to server. */
+  public void greet(String name) {
+    logger.info("Will try to greet " + name + " ...");
+    HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+    HelloReply response;
+    try {
+      response = blockingStub.sayHello(request);
+    } catch (StatusRuntimeException e) {
+      logger.log(Level.WARNING, "RPC failed: {0}", e.getStatus());
+      return;
+    }
+    logger.info("Greeting: " + response.getMessage());
+  }
+
+  /**
+   * Greet server. If provided, the second element of {@code args} is the name to use in the
+   * greeting.
+   */
+  public static void main(String[] args) throws Exception {
+    String user = "xds-client";
+    if (args.length < 1 || args.length > 3) {
+      System.out.println("USAGE: HelloWorldClientXds target [name [y]]\n");
+      System.err.println("  target  The xds target to connect to using the 'xds:' target scheme.");
+      System.err.println("  name    The name you wish to include in the greeting request. Defaults to " + user);
+      System.err.println(
+          "  'y'     'y' or 'yes' indicates using xDS credentials otherwise defaults to insecure.");
+      System.exit(1);
+    }
+    boolean useXdsCreds = false;
```

Comment:
nit: a (slightly) better style is to declare all the variables that the command line argument parsing logic will parse to at the top. This makes the main cleaner. (Well, this comment might be overly strictly...)